### PR TITLE
Add support for adc mcp3425

### PR DESCRIPTION
**Pull request by alva-engineering**
Adding support for the Microchip MCP3425 which is a 16-bit Analog-to-Digital converter with I2C interface and on-board reference

### CODE REVIEW

**Comment by alva-engineering** [comment]
@anangl Hey, i think everything is in order now? Can you let me know what else is needed for a review?

**Review by anangl — COMMENTED** [review]
You should also add a corresponding dts node in the `build_all/drivers/adc` test so that this driver is built by CI. A node like this one:
https://github.com/zephyrproject-rtos/zephyr/blob/db00177e2e7778bd4a93e16cddf5219361302fda/tests/drivers/build_all/adc/app.overlay#L85-L89

File: drivers/adc/Kconfig.mcp3425 [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+# Copyright (c) 2023 Alva Technology
+# SPDX-License-Identifier: Apache-2.0
+
+config ADC_MCP3425
+	bool "MCP3425 driver"
+	default y
+	depends on DT_HAS_MICROCHIP_MCP3425_ENABLED
+	select I2C
+	select ADC_CONFIGURABLE_INPUTS
```
[the remark this answers is not in the document]
Comment (alva-engineering) [reply drivers/adc/Kconfig.mcp3425:11]: It was indeed to get the differential field set, the current way to do that is not that transparent. I do like your proposition but it is not in main yet. How do you want me to approach this?

[the remark this answers is not in the document]
Comment (anangl) [reply drivers/adc/Kconfig.mcp3425:11]: You can cherry-pick that commit into this PR and then modify your commits accordingly ("drivers: adc: add support for MCP3425" and "dts: bindings: add bindings for MCP3425" could be also squashed into one on this occasion).

**Comment by alva-engineering** [comment]
@anangl Hi, I just came back from my summer holidays. Before i left i cherry-picked your commit and I now fixed the commit with the necessary changes. However i get a unit-test result error on the  tests/drivers/build_all/adc/drivers.adc.stm32.build (disco_l475_iot1:tests/drivers/build_all/adc/drivers.adc.stm32.build) with error : _scripts/build/gen_kobject_placeholders.py did not reserve enough space for kobject rodata._  I did not get the error locally, so i am not sure if this was because of my change?

File: drivers/adc/Kconfig.mcp3425 (hunk 1 above)
[the remark this answers is not in the document]
Comment (alva-engineering) [reply drivers/adc/Kconfig.mcp3425]: @anangl Would you be able to look at my last remark and last update of the code?

**Comment by MaureenHelm** [comment]
Can you try rebasing please?

**Pull request closed without merging**